### PR TITLE
Documentation for updown

### LINKED ISSUE #159: Missing documentation for up/down keys
It is not available in the user guide.

### PR DESCRIPTION
**Pull request by austinsantoso**
added documentation for up down
fixes #159

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -249,6 +249,114 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
...
+outcome, but the index will stay at -1.
+
+When the down key is pressed, the index will increment by 1 and the user inputted command at that index will be returned.
+If the index is currently the last possible value, the last user inputted command, pressing the down key will cause the
+index to increment by 1, which is beyond the index of the list, the empty string will be returned. Any further down key press
+will have the same outcome but the index will stay the list size or last possible index + 1.
+
+Given below is an example usage scenario and how the up/down mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. The `commandHistory` will be initialized with an empty storage ,
```
Comment (choongyx) [inline docs/DeveloperGuide.adoc]: Could use 1, 2 ,3 instead to be consistent with the rest of the DG

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -249,6 +249,114 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
...
+
+The following sequence diagram shows how the up/down key mechanism works:
+
+image::UpDownSequenceDiagram.png[]
+`getPrevCommand()` works the same way as the the sequence diagram above for `getNextCommand()`.
+The only difference is which commands get returned.
+
+==== Design Considerations
+
+===== Aspect: How undo & redo executes
```
Comment (choongyx) [inline docs/DeveloperGuide.adoc]: Not undo and redo

Comment (austinsantoso) [reply docs/DeveloperGuide.adoc]: oops

**Review by Nanosync — APPROVED** [review]

**Comment by Nanosync** [comment]
Will merge first, please fix the styling in another PR

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -249,6 +249,114 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
 
 Certain properties of the application can be controlled (e.g user prefs file location, logging level) through the configuration file (default: `config.json`).
 
+//tag:updown[]
+[[Implementation-commandHistory]]
+=== command History
```
Comment (Nanosync) [inline docs/DeveloperGuide.adoc:254]: here

**Pull request merged**